packages/api/src/routers/api/alerts.ts

```diff
@@ -43,6 +48,29 @@ const zAlert = z
 
 const zAlertInput = zAlert;
 
+const getHistory = async (alert: IAlert, teamId: string) => {
+  const histories = await AlertHistory.find({ alert: alert._id, team: teamId })
+    .sort({ createdAt: -1 })
+    .limit(20);
+  return histories;
+};
+
+const getDashboard = async (alert: IAlert, teamId: string) => {
+  const dashboard = await Dashboard.findOne({
+    _id: alert.dashboardId,
+    team: teamId,
+  });
+  return dashboard;
+};
+
+const getLogView = async (alert: IAlert, teamId: string) => {
+  const logView = await LogView.findOne({
+    _id: alert.logView,
+    team: teamId,
+  });
+  return logView;
+};
```

Comment:
No need to add these methods. You can use `populate` method. check out the example https://github.com/DeploySentinel/hyperdx-ee/blob/7886b3970f9b0745d0e74e3de9f08eb37aad6fbf/packages/api/src/tasks/checkAlerts.ts#L32-L40

Reply:
I haven't been able to get this to work, is it cool if I leave it for a follow-up refactoring?

Reply:
yap, that's fine